### DISCUSSION TITLE
unit conversion not working on stock and purchase in profit report

### DISCUSSION THREAD

**Discussion by abedfatafta** [Blair2004/NexoPOS#477] [Ideas]
**Describe the bug**
lets say we have base unit of 1 and carton of 12 and when we make procurement we order 0.5 carton , and when we sell we sell base unit  it show stock (0) and in profit report purchase price  stored as zero 

**To Reproduce**
define 2 units base and full unit 
buy full unit and sell base unit 
go to profit report 
**Expected behavior**
to convert purchase price between units . 

**Screenshots**

profit loss report 
![image](https://user-images.githubusercontent.com/16276919/148778014-23ac24df-7521-49f3-a4e6-5a7990ecb448.png)
procurement invoice 
![image](https://user-images.githubusercontent.com/16276919/148778128-4513571c-1f5d-471b-9f7f-f3889c4ed844.png)
order screenshot
![image](https://user-images.githubusercontent.com/16276919/148778232-b1270307-35d2-459c-bbe7-d7d17088e3b7.png)

**Environment:**
 - NexoPOS 4.6.8

**Comment by josephelkhoury** [comment]
Hello,

I'm facing a similar issue.

Basically if I add a new unit group for Wine bottles and then I create 2 units: Bottle, Glass.
Glass being the base unit with a value of 1, and Bottle with a value of 8 (8 glasses make 1 bottle).

If I do a procurement of 1 bottle and try to sell 1 glass, it says that the stock is 0.
Shouldn't the system calculate that 1 bottle = 8 glasses automatically?
Isn't it how the units groups and units work?

Thanks.

@Blair2004
